src/videojs-hls.js

```diff
@@ -995,7 +1019,7 @@ videojs.HlsHandler.prototype.loadSegment = function(segmentInfo) {
       return;
     }
 
-    self.lastSegmentLoaded_ = segmentInfo.uri;
+    self.lastSegmentLoaded_ = segment;
```

Comment:
I've added the byterange comparison, merged the development branch and @tenacex's changes (thanks for adding them!), and pushed them to [my fork](https://github.com/photoshelter/videojs-contrib-hls/tree/feature/byterange-support-dev). I can't update this PR since I don't have push access to @tenacex's fork. Should I create (yet another) one?